src/components/User/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useContext, useState } from 'react'
+import firebase from 'firebase'
+import { authentication } from '../../firebase'
+
+type UserType = {
+	user: firebase.User | null
+	isLoggedIn: boolean
+}
+
+export const userContext = createContext<UserType>({
+	user: null,
+	isLoggedIn: false,
+})
+
+const UserProvider: React.FC = ({ children }) => {
+	const [user, setUser] = useState(authentication.currentUser)
+	authentication.onAuthStateChanged((user) => setUser(user))
+	const isLoggedIn = user ? true : false
```

Comment:
You can convert user directly to `Boolean` type. :wink: 

```js
Boolean(null) // false
Boolean({ name: "Hello" }) // true
```

```suggestion
	const isLoggedIn = Boolean(user)
```